### DISCUSSION TITLE
Error calling WFSClient$new

### DISCUSSION THREAD

**Discussion by bwcompton** [eblondel/ows4R#108] [OGC WFS] [answered]
I'm using the following call on a GeoServer:
```
library(ows4R)
request <- 'https://umassdsl.webgis1.com/geoserver/wfs?service=wfs&request=GetCapabilities'
info <- WFSClient$new(request, serviceVersion = '2.0.0', logger = 'DEBUG')
```
And I get this error:
```
Error in names(namespaces) <- rep("", length(namespaces)) : 
  attempt to set an attribute on NULL
```
Everything works fine when I use a different server. Pasting the request into the browser gives a whole lot of XML, so there's access. The server is on AcuGIS, running GeoServer WFS v. 1.0.0, 1.1.0, and 2.0.0. I get the error with all three versions.

**Comment by eblondel** [comment, marked as answer]
Hello @bwcompton the url should only be the base URL for the service, ie: `https://umassdsl.webgis1.com/geoserver/wfs` or eventually `https://umassdsl.webgis1.com/geoserver/ows?service=WFS` (using the global OWS - OGC Web Service -endpoint)

Check this, it works:

```r
library(ows4R)
request <- 'https://umassdsl.webgis1.com/geoserver/wfs'
info <- WFSClient$new(request, serviceVersion = '2.0.0', logger = 'DEBUG')

#list feature types
fts = info$capabilities$getFeatureTypes(pretty = T)

#get features example
sf_test = info$getFeatures("ne:disputed_areas")
```

BTW, note that although it would be recommended to use WFS version `2.0.0`, GML types MultiSurface (handled through WFS 2.0) are then not well supported in the `sf` package. You can use lower version to stick with MULTIPOLYGON types, or  eventually do cast geometry types:

```r
library(magrittr)
sf_test2 = sf::st_cast(sf_test, "GEOMETRYCOLLECTION") %>% sf::st_collection_extract("POLYGON")
```

**Reply by bwcompton** [reply]
Got it--I now see where I went wrong. The server that worked for me was tolerant of my ill-formed request; GeoServer is not.

Thanks for helping push me up the learning curve--I'm still at the steep part.

**Reply by eblondel** [reply]
Great, feel free to ask more questions here if you have issues.
